packages/flutter_tools/lib/src/base/net.dart

```diff
@@ -29,7 +29,11 @@ Future<List<int>> fetchUrl(Uri url) async {
   }
 }
 
-Future<List<int>> _attempt(Uri url) async {
+/// Check if the given URL points to a valid endpoint.
+Future<bool> doesRemoteFileExist(Uri url) async =>
+  (await _attempt(url, true)) as bool;
+
+Future<dynamic> _attempt(Uri url, bool onlyHeaders) async {
```

Comment:
Don't do it this way: we try and avoid dynamic if we can, since it can make hairy type situations show up.

I'd suggest returning an empty list if the result is a 200, and null if it isn't.  Then you can check for null without having the type coercion in "doesRemoteFileExist".

Reply:
Ah, that's a good point. I figured since `_attempt` is only used in the context of this file and these two methods that confusing type situations shouldn't happen. Done.